odbc_fdw.c

```diff
@@ -135,11 +149,24 @@ static void odbcEstimateCosts(PlannerInfo *root, RelOptInfo *baserel, Cost *star
 static void odbcGetForeignPaths(PlannerInfo *root, RelOptInfo *baserel, Oid foreigntableid);
 static bool odbcAnalyzeForeignTable(Relation relation, AcquireSampleRowsFunc *func, BlockNumber *totalpages);
 static ForeignScan* odbcGetForeignPlan(PlannerInfo *root, RelOptInfo *baserel, Oid foreigntableid, ForeignPath *best_path, List *tlist, List *scan_clauses, Plan *outer_plan);
+List* odbcImportForeignSchema(ImportForeignSchemaStmt *stmt, Oid serverOid);
 
 /*
  * helper functions
  */
 static bool odbcIsValidOption(const char *option, Oid context);
+static void check_return(SQLRETURN ret, char *msg, SQLHANDLE handle, SQLSMALLINT type);
+static void normalize_empty_string(char **str);
+static char* empty_string_if_null(char *string);
+static void extract_odbcFdwOptions(List *options_list, odbcFdwOptions *extracted_options);
+void init_odbcFdwOptions(odbcFdwOptions* options);
+void copy_odbcFdwOptions(odbcFdwOptions* to, odbcFdwOptions* from);
+static void odbc_connection(odbcFdwOptions* options, SQLHENV *env, SQLHDBC *dbc);
+static void sql_data_type(SQLSMALLINT odbc_data_type, SQLULEN column_size, SQLSMALLINT decimal_digits, SQLSMALLINT nullable, StringInfo sql_type);
+static void odbcGetOptions(Oid server_oid, List *add_options, odbcFdwOptions *extracted_options);
+static void odbcGetTableOptions(Oid foreigntableid, odbcFdwOptions *extracted_options);
+static void check_return(SQLRETURN ret, char *msg, SQLHANDLE handle, SQLSMALLINT type);
+static void odbcConnStr(StringInfoData *conn_str, odbcFdwOptions* options);
```

Comment:
Actually you don't need to do any forward declaration of static ("private") functions unless you use them before actually defining them. I don't know if the style guide says something about that.